src/qiskit_iqm/iqm_job.py

```diff
@@ -103,8 +111,19 @@ def submit(self):
             'checking the progress and retrieving the results of the submitted job.'
         )
 
-    def cancel(self):
-        raise NotImplementedError('Canceling jobs is currently not supported.')
+    def cancel(self) -> bool:
+        """Attempt to cancel the job.
+
+        Returns:
+            True if the job was cancelled successfully, False otherwise
+        """
+        try:
+            self._client.abort_job(uuid.UUID(self._job_id))
+            print(f'Job {self._job_id} cancelled successfully.')
```

Comment:
We should not use `print`. In the success case I would print nothing and in the failure case I would use the `warning` module.

Reply:
Using `warnings.warn` instead of `print` now, also checking in unit tests that warning is produced when expected.